yt/geometry/_selection_routines/region_selector.pxi

```diff
@@ -41,42 +41,23 @@ cdef class RegionSelector(SelectorObject):
                         i, region_width[i]))
 
         for i in range(3):
+            # First, we check if any criteria requires a period check,
```

Comment:
@jzuhone can you look at this in light of your recent changes?

Reply:
@mameehan5 could you explain a bit why these changes were needed? we just dealt with some long-standing issues in this particular class and I want to make sure we're not accidentally breaking anything

Reply:
@jzuhone these changes were all mostly to bypass the `RuntimeError` below. Do you remember what this issues was? Or, do you possibly have an example particle dataset you are worried about? I don't see any issues on my end with my dataset, but I do know it is only a small subset of the various types of datasets yt needs to handle.